temporalio/client.py

```diff
@@ -209,15 +219,22 @@ def __init__(
             self._impl = interceptor.intercept_client(self._impl)
```

Comment:
Let's move this interceptor work to _after_ the plugin run and make sure we're using the interceptors on the config instead of the ones passed in (plugins can add interceptors).